src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -1045,6 +1049,28 @@ ErrorIfModifyQueryNotSupported(Query *queryTree)
 	foreach(rangeTableCell, rangeTableList)
 	{
 		RangeTblEntry *rangeTableEntry = (RangeTblEntry *) lfirst(rangeTableCell);
+		bool referenceTable = false;
+
+		if (IsDistributedTable(rangeTableEntry->relid))
+		{
+			DistTableCacheEntry *distTableEntry =
+				DistributedTableCacheEntry(rangeTableEntry->relid);
+
+			if (distTableEntry->replicationModel == REPLICATION_MODEL_2PC)
```

Comment:
> I believe all 2PC distributed tables should receive the modifications 

Not really. For instance, we could consider using 2PC for single shard modifications which could be safe to run from anywhere. So, I think we should use `partitionMethod` here and in case someone implements 2PC for some other modifications, let those devs decide on supporting from schema node or not.